### PR TITLE
Add maven build support for `examples`.

### PR DESCRIPTION
**Pull request by HailongWen**
This PR adds a `pom.xml` in examples to enable support for maven build.

Currently I only add `appassembler` plugin, which is an equivalent for `application` plugin in gradle and it has [MIT license](http://www.mojohaus.org/appassembler/appassembler-maven-plugin/license.html).

### CODE REVIEW

File: examples/pom.xml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            <source>1.8</source>
+            <target>1.8</target>
+          </configuration>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
```
Comment (sebright) [inline examples/pom.xml]: I don't think we need to add Checkstyle to more than one build file, since we only need to check the licenses once in CI.

Comment (HailongWen) [reply examples/pom.xml]: Done.

**Comment by HailongWen** [comment]
Subitem of #855, depending on #890 to merge.

**Comment by sebright** [comment]
@HailongWen Could you please rebase this branch onto master?  It looks like the diff still contains #890.

**Comment by bogdandrutu** [comment]
You need to use rebase when sync your branch with the master, not merge. Please also consider to use "squash and merge" when merge to the master.

File: examples/pom.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  <artifactId>opencensus-examples</artifactId>
+  <packaging>jar</packaging>
+  <version>0.11.0-SNAPSHOT</version><!-- CURRENT_OPENCENSUS_VERSION -->
+  <name>opencensus-examples</name>
+  <url>http://maven.apache.org</url>
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <!-- change to the version you want to use. -->
+    <opencensus.version>0.10.1</opencensus.version><!-- LATEST_OPENCENSUS_VERSION -->
+    <checkstyle.failsOnError>false</checkstyle.failsOnError>
```
Comment (sebright) [inline examples/pom.xml]: This property can also be removed.

Comment (HailongWen) [reply examples/pom.xml]: Done.

**Pull request merged**